flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/SnapshotFinder.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.file.utils;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.core.fs.FileStatus;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.UUID;
+import java.util.function.BinaryOperator;
+
+/** Find latest and earliest snapshot. */
+public class SnapshotFinder {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SnapshotFinder.class);
+
+    public static final String SNAPSHOT_PREFIX = "snapshot-";
+
+    public static final String EARLIEST = "EARLIEST";
+
+    public static final String LATEST = "LATEST";
+
+    public static Long findLatest(Path snapshotDir) throws IOException {
+        FileSystem fs = snapshotDir.getFileSystem();
+        if (!fs.exists(snapshotDir)) {
+            return null;
+        }
+
+        try {
+            long snapshotId = readHint(snapshotDir, LATEST);
+            long nextSnapshot = snapshotId + 1;
+            // it is the latest only there is no next one
+            if (!fs.exists(new Path(snapshotDir, SNAPSHOT_PREFIX + nextSnapshot))) {
+                return snapshotId;
+            }
+        } catch (Exception ignore) {
```

Comment:
Add a warning log.

Reply:
changed to exists instead of exception catching.